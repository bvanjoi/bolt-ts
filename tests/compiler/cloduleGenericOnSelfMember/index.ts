// From `github.com/microsoft/TypeScript/blob/v5.8.2/tests/cases/compiler/cloduleGenericOnSelfMember.ts`, Apache-2.0 License

class ServiceBase<T> {
  field: T;
}
class Service extends ServiceBase<typeof Service.Base> {
}
namespace Service {
  export const Base = {
      name: "1",
      value: 5
  };
}

function f(a: ServiceBase<typeof Service.Base>) {
  let b: number = a.field.name;
  //~^ ERROR: Type 'string' is not assignable to type 'number'.
}
