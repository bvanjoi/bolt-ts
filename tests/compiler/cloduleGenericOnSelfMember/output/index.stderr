  × Type 'string' is not assignable to type 'number'.
    ╭─[../../tests/compiler/cloduleGenericOnSelfMember/index.ts:16:7]
 16 │ b
    · ─
    ╰────
